app/ui/legacy/src/main/java/com/fsck/k9/activity/AccountList.kt

```diff
@@ -0,0 +1,151 @@
+package com.fsck.k9.activity
+
+import android.os.Bundle
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemClickListener
+import android.widget.ArrayAdapter
+import com.fsck.k9.Account
+import com.fsck.k9.BaseAccount
+import com.fsck.k9.K9.isShowUnifiedInbox
+import com.fsck.k9.Preferences.Companion.getPreferences
+import com.fsck.k9.search.SearchAccount.Companion.createUnifiedInboxAccount
+import com.fsck.k9.ui.R
+import com.google.android.material.textview.MaterialTextView
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ * Activity displaying the list of accounts.
+ *
+ * Classes extending this abstract class have to provide an [.onAccountSelected]
+ * method to perform an action when an account is selected.
+ */
+abstract class AccountList : K9ListActivity(), OnItemClickListener {
+
+    private val scope = MainScope()
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setResult(RESULT_CANCELED)
+        setLayout(R.layout.account_list)
+
+        listView.apply {
+            onItemClickListener = this@AccountList
+            itemsCanFocus = false
+        }
+    }
+
+    /**
+     * Reload list of accounts when this activity is resumed.
+     */
+    public override fun onResume() {
+        super.onResume()
+        loadAccounts()
+    }
+
+    /**
+     * Cancel all running coroutines when this activity is paused.
+     */
+    public override fun onPause() {
+        super.onPause()
+        scope.cancel()
```

Comment:
[Bug] Once the activity is paused the scope is canceled. Resuming the activity won't lead to the account list being reloaded because launching a coroutine on a canceled scope does nothing.

Reply:
This might be a bug if it would use a regular scope. This is using a `MainScope`, that is composed of `SupervisorJob()` + `Dispatchers.Main`, which prevents exactly this issue.

Reply:
This is not canceling a child of the scope, but the scope itself. I did test this on a device and pausing, then resuming didn't refresh the account list like it should.

Reply:
Hmm, strange I just put a timber log in and it works

Reply:
But I'll update to the Lifecycle version, too keep it simple